### PR TITLE
Feature/server hooks

### PR DESCRIPTION
**Pull request by paulsouche**

### CODE REVIEW

File: packages/git-hook/src/Hook.ts [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
+export default abstract class Hook {
     public static readonly ZERO_COMMIT = '0000000000000000000000000000000000000000';
 
+    public readonly config: object;
     public readonly oldRev: string;
     public readonly newRev: string;
     public readonly refName: string;
 
-    constructor() {
+    constructor(config: object) {
```
Comment (lsagetlethias) [inline packages/git-hook/src/Hook.ts]: `constructor(public readonly config: object) {` ?

File: packages/git-hook/src/git/GetNewCommitsList.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    if (!matches) {
+        return;
+    }
+
+    return matches.pop();
+}
+
+async function parseCommit(rawCommit: string): Promise<Commit> {
+    const rawParse = await gitCommand(`cat-file -p ${rawCommit}`);
```
Comment (lsagetlethias) [inline packages/git-hook/src/git/GetNewCommitsList.ts]: non reproductible in ts ?

**Review by lsagetlethias — DISMISSED** [review]

File: packages/git-hook/src/git/GetNewCommitsList.ts (hunk 2 above)
Comment (paulsouche) [reply packages/git-hook/src/git/GetNewCommitsList.ts]: Meaning without dependency ?

Comment (lsagetlethias) [reply packages/git-hook/src/git/GetNewCommitsList.ts]: Resolved as we spoke.

File: packages/git-hook/src/git/GetNewCommitsList.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import gitCommand from './GitCommand';
```
Comment (easyni) [inline packages/git-hook/src/git/GetNewCommitsList.ts]: why uppercase on file name ? You don't  expose a class ?

File: packages/git-hook/src/git/GitCommand.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import { exec } from 'child_process';
```
Comment (easyni) [inline packages/git-hook/src/git/GitCommand.ts]: why uppercase on file name ? You don't expose a class ?

**Review by easyni — COMMENTED** [review]
OK for only some question about file's naming.

**Review by easyni — APPROVED** [review]

**Pull request merged**